File: algorithms/1984. Minimum Difference Between Highest and Lowest of K Scores/solution.py
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        """Sort then use slide window."""
        nums.sort()
        m = 100000
        for i in range(k - 1, len(nums)):
            m = min(m, nums[i] - nums[i - k + 1])
        return m

File: algorithms/1984. Minimum Difference Between Highest and Lowest of K Scores/solution.go
func minimumDifference(nums []int, k int) int {
    if k == 1 {
        return 0
    }
    sort.Ints(nums)
    m := 100000
    for i := k - 1; i < len(nums); i++ {
        if m > nums[i] - nums[i-k+1] {
            m = nums[i] - nums[i-k+1]
        }
    }
    return m
}
